src/services/infrastructure/entity-resolver/storage.bucket.resolver.service.ts

```diff
@@ -1,32 +1,45 @@
 import { Inject, Injectable, LoggerService } from '@nestjs/common';
 import { LogContext } from '@common/enums';
-import { EntityManager } from 'typeorm';
+import { EntityManager, ObjectType } from 'typeorm';
 import { InjectEntityManager } from '@nestjs/typeorm';
 import { StorageBucketNotFoundException } from '@common/exceptions/storage.bucket.not.found.exception';
 import { WINSTON_MODULE_NEST_PROVIDER } from 'nest-winston';
+import { replaceRegex } from '@common/utils/replace.regex';
+import { Reference } from '@domain/common/reference';
+import { BaseAlkemioEntity } from '@domain/common/entity/base-entity';
+import { AgentInfo } from '@core/authentication';
+import { fromBuffer } from 'file-type';
+import { IpfsService } from '@services/adapters/ipfs/ipfs.service';
+import { StorageBucketService } from '@domain/storage/storage-bucket/storage.bucket.service';
+import { MimeTypeNotFoundException } from '@common/exceptions/mime.type.not.found.exception';
+import { Visual } from '@domain/common/visual';
 
 @Injectable()
 export class StorageBucketResolverService {
   constructor(
     @InjectEntityManager('default')
     private entityManager: EntityManager,
-    @Inject(WINSTON_MODULE_NEST_PROVIDER) private readonly logger: LoggerService
+    @Inject(WINSTON_MODULE_NEST_PROVIDER)
+    private readonly logger: LoggerService,
+    private ipfsService: IpfsService,
+    private storageBucketService: StorageBucketService
   ) {}
 
   public async getStorageBucketIdForProfile(
     profileID: string
-  ): Promise<string> {
+  ): Promise<string | never> {
     // First iterate over all the entity types that have storage spaces directly
     for (const entityName of Object.values(DirectStorageBucketEntityType)) {
-      const match = await this.getDirectStorageBucketForProfile(
+      const match = await getDirectStorageBucketForProfile(
```

Comment:
what is the rational for moving the functions out of the class?

Reply:
The rational was that there were two ways to go about this one-off (my understanding) migration:

1. Quick and dirty code, develop as quickly as possible, test, deploy, execute, and revert the unnecessary code. No additional authorization etc. needed, as this will be executed once, and executing it twice won't actually change anything. 
2. Spend a little more time and polish it up - easier to review, better code, looks better. Takes more time.

My understanding provided the feedback around InnovationSpaces, is we prefer `1` over `2`. That was my reasoning. Regardless, I have already adjusted the B branch and merged it in this one, which reverts to `2`, and pretty much addresses all comments.

Reply:
I know that was the steer, and we do need to get this out - but it was fairly quick for me to move the code so that it was aat least isolated.